### ISSUE TITLE
Please update all dependencies

### ISSUE THREAD

**Issue by lepture** [webpack/webpack-dev-server#260]
It can't work well on node 4.0

**Comment by andresmrm** [comment]
Getting `segmentation fault (core dumped)  webpack-dev-server` with nodejs 4.0.
Downgraded to nodejs 0.12.7 and it's working again.

**Comment by sokra** [comment]
have you tried `npm rebuild`

**Comment by razagill** [comment]
I'm getting the same error after updating to node 4.0. Also tried `npm rebuild` but still the same.

**Comment by razagill** [comment]
Deleted the `~/.node-gyp` folder and then did a `npm rebuild` again which has seemed to work for now. Although I still errors upon installing `webpack-dev-server`. Same as #258

**Comment by lepture** [comment]
@razagill coz socket.io depends on an old ws.

**Comment by CrocoDillon** [comment]
Seems like [socket.io](https://github.com/socketio/socket.io/issues/2236) is still not updated for Node 4 though, which is sad. Anyway tagging along to get notified when this is fixed.

**Comment by jinishiu** [comment]
same problem caused by socket.io

**Comment by ChALkeR** [comment]
Adding to the list: https://github.com/nodejs/node/issues/2798.

**Comment by eugene1g** [comment]
webpack-dev-server depends on two socket.io packages that have just been patched to work on iojs3 or node4. They haven't cut new releases yet, but you can change the webpack-dev-server package.json file to use the patched versions directly like so 
```
    "socket.io": "socketio/socket.io#210e6887327e6afed4761940fae405b8e784e8e1",
    "socket.io-client": "socketio/socket.io-client#4c22f33c0b07e9eb21a65e342581506a1c44072b",
```
Of course this hack isn't a lasting solution and can't be done in production, but for now it's a simple workaround to get the dev-server stuff working again in your dev environment.

**Comment by mattgi** [comment]
Need a `socket.io` fork that also points its dependency of `socket.io-client` to what you have provided @eugene1g.

So it needs to be:
```
"socket.io": "socketio/socket.io#f5b753",
"socket.io-client": "socketio/socket.io-client#4c22f3",
```

f5b753 being a later commit.

**Comment by SpaceK33z** [comment]
A new version of socket.io is just released which includes the fix to work on nodejs v4 (`1.3.7`). I will write a PR for this.

**Comment by chemzqm** [comment]
Build errors still exist, but only raise on optional deps [https://github.com/BrowserSync/browser-sync/issues/820]. Looks like no impact on daily usage, just ignore them would be fine.
